### PR TITLE
Warnint util 4516 v7.2

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/4516

Describe changes:
- Fix integer warnings `-Wimplicit-int-conversion` for util files

Seems to be uncontroversial part of #7006 as #7081 was
Plus fixes the new warnings introduced meanwhile, like one introduced in https://github.com/OISF/suricata/pull/7052 and the radix changes

Replaces #7100 with CIFuzz fix for util-proto-name (aka how to make a complete loop for every uint8_t value)

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 6463

File: src/util-spm-bm.c [hunk 1]
```diff
@@ -267,11 +270,11 @@ static void PreBmGsNocase(const uint8_t *x, uint16_t m, uint16_t *bmGs)
         bmGs[i] = m;
     }
     j = 0;
-    for (i = m - 1; i >= 0; --i) {
-        if (i == -1 || suff[i] == i + 1) {
-            for (; j < m - 1 - i; ++j) {
+    for (i = m; i > 0; --i) {
```
Comment (victorjulien) [inline src/util-spm-bm.c:273]: this looks like the only logic change, and I have a failing test that only happens with this branch + spm-algo=bm, while it works in master or in this branch with spm-algo=hs

Will share the test offline

Comment (catenacyber) [reply src/util-spm-bm.c:273]: indeed, I see the bug

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/7107

File: src/util-spm-bm.c (hunk 1 above)
Comment (catenacyber) [reply src/util-spm-bm.c:273]: I changed every usage of `i` by 1 except the one `suff[i]`

**Pull request closed without merging**